The use of RTLD_DEEPBIND is just a workaround and not a real fix.
And we cannot use it...

--- ./src/effects/VST/VSTEffect.cpp.orig
+++ ./src/effects/VST/VSTEffect.cpp
@@ -2609,6 +2609,10 @@
    // symbols.
    //
    // Once we define a proper external API, the flags can be removed.
+// GNU extension
+#ifndef RTLD_DEEPBIND
+#define RTLD_DEEPBIND 0
+#endif
    void *lib = dlopen((const char *)wxString(realPath).ToUTF8(), RTLD_NOW | RTLD_LOCAL | RTLD_DEEPBIND);
    if (!lib) 
    {
